Makefile

```diff
@@ -114,20 +116,24 @@ e2e.test:
 test-e2e: e2e.test
 	./e2e.test -test.timeout 180m -test.v -ginkgo.v
 
-test-go: generate
-	go build -tags containers_image_openpgp ./...
+test-go: generate build-all validate-go lint-go unit-test-go coverage-go
```

Comment:
1. As far as I see we currently (in `master`) do not run `lint-go` as part of `test-go` before. You think we should? It is quite a slow target: might be a good idea to leave it for CI.
2. It looks like both `unit-test-go` and `coverage-go` run all unit tests: they just generate different reports: as a result `Golang_Unit_Tests` job takes 14 minutes instead of 7-9 minutes compared to `master` version of `Makefile` / pipeline config.

Reply:
> I am no longer running `test-go` in our pipeline, I have split the code.

Yep, I understnad about spliting code in the pipeline. It was more about having `lint-go` as part of `test-go`.

> I believe it is for the best if `lint-go` is part of test-go as it will warn early when something is fishy in the code. I personally hate to find fail on the ci, that I could have easily fixed.

Yeah, I'm kind on the fence with this... I personally never run `test-go` locally: I only run specific relevant tests and let CI do the rest. So it doesn't make any difference for my workflow, but I realise that other people probably do use it locally.

Let's put it into `test-go` and see how people feel about it. We can change it, if it causes a lot of pain.